Document `headingoffset` and `headingreset` HTML global attributes (#45010)
mdn/content: files/en-us/web/html/reference/global_attributes/headingreset/index.md

---
title: "`headingreset` HTML global attribute"
short-title: headingreset
slug: Web/HTML/Reference/Global_attributes/headingreset
page-type: html-attribute

> sidebar: htmlsidebar

Reviewer: Please restore your original version that had the `status` and `browser-compat` keys

---

{{SeeCompatTable}}

The **`headingreset`** [global attribute](/en-US/docs/Web/HTML/Reference/Global_attributes) stops the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values of ancestor elements from applying to the [heading elements](/en-US/docs/Web/HTML/Reference/Elements/Heading_Elements) inside the element it is set on.

## Values

> This is a [boolean attribute](/en-US/docs/Glossary/Boolean/HTML). Its presence enables it. The value is ignored, so `headingreset`, `headingreset=""`, and `headingreset="false"` all have the same effect.

Reviewer: ```suggestion
This is a [boolean attribute](/en-US/docs/Glossary/Boolean/HTML); it takes effect when it is present. A value for it is ignored, so `headingreset`, `headingreset=""`, and `headingreset="false"` all have the same effect.
```

page-type: html-attribute
status:
  - experimental
browser-compat: html.global_attributes.headingreset
sidebar: htmlsidebar
---

{{SeeCompatTable}}

The **`headingreset`** [global attribute](/en-US/docs/Web/HTML/Reference/Global_attributes) stops the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values of ancestor elements from applying to the [heading elements](/en-US/docs/Web/HTML/Reference/Elements/Heading_Elements) inside the element it is set on.

## Values

This is a [boolean attribute](/en-US/docs/Glossary/Boolean/HTML); it takes effect when it is present. A value for it is ignored, so `headingreset`, `headingreset=""`, and `headingreset="false"` all have the same effect.

## Description

When a browser computes the [heading level](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset#description) of a heading element, it walks up the ancestor chain and adds up the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values it finds. The `headingreset` attribute stops that walk: offsets set on the element itself still count, but no ancestor above it contributes an offset.

> Use this attribute for content that is not part of the structure of the surrounding document, such as a {{htmlelement("dialog")}}, a popover, or markup inserted from another source. Their headings then start again at level 1.

Reviewer: This should probably be corrected to:
```suggestion
Use this attribute for content that is not part of the structure of the surrounding document, such as a {{htmlelement("dialog")}}, a popover, or markup inserted from another source. Their headings then take the level of their element name.
```

---
title: "`headingreset` HTML global attribute"
short-title: headingreset
slug: Web/HTML/Reference/Global_attributes/headingreset
page-type: html-attribute
status:
  - experimental
browser-compat: html.global_attributes.headingreset
sidebar: htmlsidebar
---

{{SeeCompatTable}}

> The **`headingreset`** [global attribute](/en-US/docs/Web/HTML/Reference/Global_attributes) is a [boolean attribute](/en-US/docs/Glossary/Boolean/HTML) that stops the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values of ancestor elements from applying to the [headings](/en-US/docs/Web/HTML/Reference/Elements/Heading_Elements) inside it.

Reviewer: ```suggestion
The **`headingreset`** [global attribute](/en-US/docs/Web/HTML/Reference/Global_attributes) stops the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values of ancestor elements from applying to the [headings](/en-US/docs/Web/HTML/Reference/Elements/Heading_Elements) inside it.
```

## Syntax

```html
<dialog headingreset>
  <h1>This heading has a computed heading level of 1</h1>
</dialog>
```

> Like other boolean attributes, its presence enables it. The value is ignored, so `headingreset`, `headingreset=""`, and `headingreset="false"` all have the same effect.

Reviewer: ```suggestion
Like other boolean attributes, `headingreset` takes effect when it is present. A value for it is ignored, so `headingreset`, `headingreset=""`, and `headingreset="false"` all have the same effect.
```

> 

Reviewer: Would be nice to have an Accessibility section on this page as well

## Description

> When a browser computes the [heading level](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset#description) of a heading element, it walks up the ancestor chain and adds up the [`headingoffset`](/en-US/docs/Web/HTML/Reference/Global_attributes/headingoffset) values it finds. The `headingreset` attribute stops that walk: offsets set on the element itself still count, but no ancestor above it contributes an offset.

Reviewer: this part "The `headingreset` attribute stops that walk: offsets set on the element itself still count, but no ancestor above it contributes an offset." on this page is correct - this explanation is missing from the `headingoffset` page

> Use it for content that is not part of the structure of the surrounding document, such as a {{htmlelement("dialog")}}, a popover, or content inserted from another source, where headings should start again at level 1.

Reviewer: ```suggestion
Use this attribute for content that is not part of the structure of the surrounding document, such as a {{htmlelement("dialog")}}, a popover, or content inserted from another source, where headings should start again at level 1.
```

If the element carries both `headingoffset` and `headingreset`, its own offset applies and everything above it is discarded.

## Examples

### Resetting heading levels in a dialog

> The `<section>` offsets its headings by 2, so its own heading is at level 3. The dialog has `headingreset`, so the heading inside it is back at level 1:

Reviewer: "back at level 1" suggests the walk and offset accumulation happened and returned to 1, but the walk was stopped at dialog's `headingreset`. Added a rephrasing suggestion:
```suggestion
In this example, the `<section>` offsets its headings by `2`, so the heading directly inside it is at level 3. The `<dialog>` element has `headingreset`, so the section's offset never reaches the heading inside the dialog, which stays at level 1:
```

Reviewer: this suggestion was not applied and I don't see a reply, so reopening the thread.

```html
<section headingoffset="2">
  <h1>Level 3</h1>

  <dialog headingreset>
    <h1>Level 1</h1>
    <p>The offset of the section does not apply here.</p>
  </dialog>
</section>
```

> ### Combining `headingreset` with `headingoffset`

Reviewer: ```suggestion
### Using `headingreset` and `headingoffset` on the same element
```

> Here the outer offset of 3 is discarded, and only the offset on the element with `headingreset` is applied:

Reviewer: ```suggestion
In this example, the outer offset of `3` never reaches the heading. Only the offset on the element with `headingreset` applies, so the heading is at level 2:
```
